jekyll/_posts/2020-10-14-what-are-arc-and-orc-in-nim.md

```diff
@@ -0,0 +1,181 @@
+(including the ones I talked about above):
+
+- **Scope-based memory management** (destructors are injected after the scope) -
+generally reduces RAM usage of the programs and improves performance
+
+- **Move semantics** - the ability for the compiler to statically analyse
+the program and convert memory copies into moves where it is possible.
+
+- **Shared heap** (different threads have access to the same memory and
+you don't need to copy variables to pass them between threads - you can instead move them;
+also see [an RFC about isolating and sending data between threads](https://github.com/nim-lang/RFCs/issues/244))
+
+- **Easier FFI** (`refc` requires each foreign thread to set up the GC
+manually which is not the case with ARC). This also means that ARC is a much better
+choice for creating Nim libraries to be used from other languages (.dll, .so, [Python extensions](https://github.com/yglukhov/nimpy), etc)
+
+- Suitable for [**hard realtime**](https://en.wikipedia.org/wiki/Real-time_computing) programming
+
+- **Copy elision** (cursor inference) reduces copies to simple cursors (aliases) in a lot of cases.
+
+Generally, ARC is an amazing step for programs to become faster, use less memory, and have predictable behaviour.
+
+To enable ARC for your program, all you have to do is compile with
+the ``--gc:arc`` switch, or add it to your project's config file (`.nims` or `.cfg`).
+
+
+
+## Problem with cycles
+
+But wait! Haven't we forgotten something? ARC is **reference counting**,
+and as we all know, RC doesn't deal with *cycles* by itself.
+In short, a cycle is when some variables depend on each other in a way that can resemble a cycle.
+Let's make a simple example: we have 3 objects (A, B, C), and each of them references the other, better shown with a diagram:
+
+<p style="text-align: center;">
+  <img width="256" height="256" src="{{ site.baseurl }}/assets/news/images/yardanico-arc/cycle.svg">
+</p>
+
+To find and collect that cycle we need to have a **cycle collector** - special part
+of the runtime which finds and removes cycles that are no longer needed in the program.
+
+In Nim cycle collection has been done by the mark & sweep phase of the `refc` GC,
+but it's better to use ARC as the foundation to make something better, so please welcome:
+
+
+## ORC - Nim's cycle collector
+ORC is an all-new Nim's cycle collector based on ARC.
+It can be considered a full-blown GC since it includes a local tracing phase
+(contrary to most other tracing GCs which do global tracing).
+I don't have much to say about ORC, but it retains most of ARC's advantages
+except **determinism** (partially) - by default ORC has an adaptive threshold
+for collecting cycles, and **hard realtime** (partially), for the same reason.
+To enable ORC you would need to compile your program with ``--gc:orc``,
+but ORC will probably become Nim's default GC in the future.
+
+
+
+## I'm excited! How do I test them out?
```

Comment:
This section will need to be changed if we plan to post the article after 1.4 is out